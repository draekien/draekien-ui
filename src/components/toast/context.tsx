import * as React from 'react';
import { ToastProps } from '.';

export type ToastContextProps = {
  toasts: ToastProps[];
  open: Function;
  close: Function;
  closeAll: Function;
  success: Function;
  error: Function;
  warning: Function;
  info: Function;
};

export const ToastContext = React.createContext<ToastContextProps>({
  toasts: [],
  open: () => {},
  close: () => {},
  closeAll: () => {},
  success: () => {},
  error: () => {},
  warning: () => {},
  info: () => {},
});

export const useToastContext = () => React.useContext(ToastContext);
