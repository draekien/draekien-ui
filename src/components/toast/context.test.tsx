/** @jsxRuntime classic */
import { ToastContext } from './context';

describe('ToastContext', () => {
  test('should match snapshot', () => {
    expect(ToastContext).toMatchSnapshot();
  });
});
