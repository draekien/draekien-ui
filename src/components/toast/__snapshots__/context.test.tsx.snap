// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastContext should match snapshot 1`] = `
Object {
  "$$typeof": Symbol(react.context),
  "Consumer": Object {
    "$$typeof": Symbol(react.context),
    "_calculateChangedBits": null,
    "_context": [Circular],
  },
  "Provider": Object {
    "$$typeof": Symbol(react.provider),
    "_context": [Circular],
  },
  "_calculateChangedBits": null,
  "_currentRenderer": null,
  "_currentRenderer2": null,
  "_currentValue": Object {
    "close": [Function],
    "closeAll": [Function],
    "error": [Function],
    "info": [Function],
    "open": [Function],
    "success": [Function],
    "toasts": Array [],
    "warning": [Function],
  },
  "_currentValue2": Object {
    "close": [Function],
    "closeAll": [Function],
    "error": [Function],
    "info": [Function],
    "open": [Function],
    "success": [Function],
    "toasts": Array [],
    "warning": [Function],
  },
  "_threadCount": 0,
}
`;
